source/server/hot_restarting_parent.cc

```diff
@@ -106,18 +106,25 @@ HotRestartingParent::Internal::getListenSocketsForChild(const HotRestartMessage:
   wrapped_reply.mutable_reply()->mutable_pass_listen_socket()->set_fd(-1);
   Network::Address::InstanceConstSharedPtr addr =
       Network::Utility::resolveUrl(request.pass_listen_socket().address());
+
   for (const auto& listener : server_->listenerManager().listeners()) {
     Network::ListenSocketFactory& socket_factory = listener.get().listenSocketFactory();
     if (*socket_factory.localAddress() == *addr && listener.get().bindToPort()) {
-      // worker_index() will default to 0 if not set which is the behavior before this field
-      // was added. Thus, this should be safe for both roll forward and roll back.
-      if (request.pass_listen_socket().worker_index() < server_->options().concurrency()) {
-        wrapped_reply.mutable_reply()->mutable_pass_listen_socket()->set_fd(
-            socket_factory.getListenSocket(request.pass_listen_socket().worker_index())
-                ->ioHandle()
-                .fdDoNotUse());
+      Network::Socket::Type socket_type =
+          Network::Utility::urlIsTcpScheme(request.pass_listen_socket().address())
+              ? Network::Socket::Type::Stream
+              : Network::Socket::Type::Datagram;
```

Comment:
It is also possible as a unix socket:
https://github.com/envoyproxy/envoy/blob/f9d936ec65e42308f72b48e52ed5b810effc40e5/source/server/listener_manager_impl.cc#L178-L187